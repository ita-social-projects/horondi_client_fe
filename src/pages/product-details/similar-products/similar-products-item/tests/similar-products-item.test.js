import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import * as reactRedux from 'react-redux';
import { faDollarSign } from '@fortawesome/free-solid-svg-icons';

import { Rating } from '@material-ui/lab';
import SimilarProductsItem from '../index';

jest.mock('@material-ui/styles', () => ({
  ...jest.requireActual('@material-ui/styles'),
  useTheme: () => ({
    palette: {
      type: 'light'
    }
  })
}));

describe('Similar products test', () => {
  let wrapper;

  beforeAll(() => {
    jest
      .spyOn(reactRedux, 'useSelector')
      .mockImplementation((selector) => selector({ Language: { language: 0 } }));

    wrapper = mount(
      <BrowserRouter>
        <SimilarProductsItem
          imageUrl='/img'
          id='some id'
          name='some name'
          rate={100}
          price={110}
          currencySign={faDollarSign}
        />
      </BrowserRouter>
    );
  });

  it('Should render similar products component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render rating component', () => {
    expect(wrapper.exists(Rating)).toBe(true);
  });
});
