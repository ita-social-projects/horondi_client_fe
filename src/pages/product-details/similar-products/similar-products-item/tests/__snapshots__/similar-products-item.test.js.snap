// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Similar products test Should render similar products component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <SimilarProductsItem
      currencySign={
        Object {
          "icon": Array [
            288,
            512,
            Array [],
            "f155",
            "M209.2 233.4l-108-31.6C88.7 198.2 80 186.5 80 173.5c0-16.3 13.2-29.5 29.5-29.5h66.3c12.2 0 24.2 3.7 34.2 10.5 6.1 4.1 14.3 3.1 19.5-2l34.8-34c7.1-6.9 6.1-18.4-1.8-24.5C238 74.8 207.4 64.1 176 64V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48h-2.5C45.8 64-5.4 118.7.5 183.6c4.2 46.1 39.4 83.6 83.8 96.6l102.5 30c12.5 3.7 21.2 15.3 21.2 28.3 0 16.3-13.2 29.5-29.5 29.5h-66.3C100 368 88 364.3 78 357.5c-6.1-4.1-14.3-3.1-19.5 2l-34.8 34c-7.1 6.9-6.1 18.4 1.8 24.5 24.5 19.2 55.1 29.9 86.5 30v48c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-48.2c46.6-.9 90.3-28.6 105.7-72.7 21.5-61.6-14.6-124.8-72.5-141.7z",
          ],
          "iconName": "dollar-sign",
          "prefix": "fas",
        }
      }
      id="some id"
      imageUrl="/img"
      name="some name"
      price={110}
      rate={100}
    >
      <Link
        to="/product/some id"
      >
        <LinkAnchor
          href="/product/some id"
          navigate={[Function]}
        >
          <a
            href="/product/some id"
            onClick={[Function]}
          >
            <div
              className="makeStyles-similarItem-1 makeStyles-similarItem-4"
            >
              <div
                className="makeStyles-info-2"
              >
                <span />
                <span>
                  110
                </span>
                <WithStyles(ForwardRef(Rating))
                  className="makeStyles-rating-3"
                  readOnly={true}
                  size="small"
                  value={100}
                >
                  <ForwardRef(Rating)
                    className="makeStyles-rating-3"
                    classes={
                      Object {
                        "decimal": "MuiRating-decimal",
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "icon": "MuiRating-icon",
                        "iconActive": "MuiRating-iconActive",
                        "iconEmpty": "MuiRating-iconEmpty",
                        "iconFilled": "MuiRating-iconFilled",
                        "iconFocus": "MuiRating-iconFocus",
                        "iconHover": "MuiRating-iconHover",
                        "label": "MuiRating-label",
                        "pristine": "MuiRating-pristine",
                        "readOnly": "MuiRating-readOnly",
                        "root": "MuiRating-root",
                        "sizeLarge": "MuiRating-sizeLarge",
                        "sizeSmall": "MuiRating-sizeSmall",
                        "visuallyhidden": "MuiRating-visuallyhidden",
                      }
                    }
                    readOnly={true}
                    size="small"
                    value={100}
                  >
                    <span
                      aria-label="100 Stars"
                      className="MuiRating-root makeStyles-rating-3 MuiRating-sizeSmall MuiRating-readOnly"
                      onMouseLeave={[Function]}
                      onMouseMove={[Function]}
                      role="img"
                    >
                      <span
                        key="1"
                      >
                        <IconContainer
                          className="MuiRating-icon MuiRating-iconFilled"
                          value={1}
                        >
                          <span
                            className="MuiRating-icon MuiRating-iconFilled"
                          >
                            <ForwardRef(StarIcon)
                              fontSize="inherit"
                            >
                              <WithStyles(ForwardRef(SvgIcon))
                                fontSize="inherit"
                              >
                                <ForwardRef(SvgIcon)
                                  classes={
                                    Object {
                                      "colorAction": "MuiSvgIcon-colorAction",
                                      "colorDisabled": "MuiSvgIcon-colorDisabled",
                                      "colorError": "MuiSvgIcon-colorError",
                                      "colorPrimary": "MuiSvgIcon-colorPrimary",
                                      "colorSecondary": "MuiSvgIcon-colorSecondary",
                                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                      "root": "MuiSvgIcon-root",
                                    }
                                  }
                                  fontSize="inherit"
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                                    focusable="false"
                                    viewBox="0 0 24 24"
                                  >
                                    <path
                                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                                    />
                                  </svg>
                                </ForwardRef(SvgIcon)>
                              </WithStyles(ForwardRef(SvgIcon))>
                            </ForwardRef(StarIcon)>
                          </span>
                        </IconContainer>
                      </span>
                      <span
                        key="2"
                      >
                        <IconContainer
                          className="MuiRating-icon MuiRating-iconFilled"
                          value={2}
                        >
                          <span
                            className="MuiRating-icon MuiRating-iconFilled"
                          >
                            <ForwardRef(StarIcon)
                              fontSize="inherit"
                            >
                              <WithStyles(ForwardRef(SvgIcon))
                                fontSize="inherit"
                              >
                                <ForwardRef(SvgIcon)
                                  classes={
                                    Object {
                                      "colorAction": "MuiSvgIcon-colorAction",
                                      "colorDisabled": "MuiSvgIcon-colorDisabled",
                                      "colorError": "MuiSvgIcon-colorError",
                                      "colorPrimary": "MuiSvgIcon-colorPrimary",
                                      "colorSecondary": "MuiSvgIcon-colorSecondary",
                                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                      "root": "MuiSvgIcon-root",
                                    }
                                  }
                                  fontSize="inherit"
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                                    focusable="false"
                                    viewBox="0 0 24 24"
                                  >
                                    <path
                                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                                    />
                                  </svg>
                                </ForwardRef(SvgIcon)>
                              </WithStyles(ForwardRef(SvgIcon))>
                            </ForwardRef(StarIcon)>
                          </span>
                        </IconContainer>
                      </span>
                      <span
                        key="3"
                      >
                        <IconContainer
                          className="MuiRating-icon MuiRating-iconFilled"
                          value={3}
                        >
                          <span
                            className="MuiRating-icon MuiRating-iconFilled"
                          >
                            <ForwardRef(StarIcon)
                              fontSize="inherit"
                            >
                              <WithStyles(ForwardRef(SvgIcon))
                                fontSize="inherit"
                              >
                                <ForwardRef(SvgIcon)
                                  classes={
                                    Object {
                                      "colorAction": "MuiSvgIcon-colorAction",
                                      "colorDisabled": "MuiSvgIcon-colorDisabled",
                                      "colorError": "MuiSvgIcon-colorError",
                                      "colorPrimary": "MuiSvgIcon-colorPrimary",
                                      "colorSecondary": "MuiSvgIcon-colorSecondary",
                                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                      "root": "MuiSvgIcon-root",
                                    }
                                  }
                                  fontSize="inherit"
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                                    focusable="false"
                                    viewBox="0 0 24 24"
                                  >
                                    <path
                                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                                    />
                                  </svg>
                                </ForwardRef(SvgIcon)>
                              </WithStyles(ForwardRef(SvgIcon))>
                            </ForwardRef(StarIcon)>
                          </span>
                        </IconContainer>
                      </span>
                      <span
                        key="4"
                      >
                        <IconContainer
                          className="MuiRating-icon MuiRating-iconFilled"
                          value={4}
                        >
                          <span
                            className="MuiRating-icon MuiRating-iconFilled"
                          >
                            <ForwardRef(StarIcon)
                              fontSize="inherit"
                            >
                              <WithStyles(ForwardRef(SvgIcon))
                                fontSize="inherit"
                              >
                                <ForwardRef(SvgIcon)
                                  classes={
                                    Object {
                                      "colorAction": "MuiSvgIcon-colorAction",
                                      "colorDisabled": "MuiSvgIcon-colorDisabled",
                                      "colorError": "MuiSvgIcon-colorError",
                                      "colorPrimary": "MuiSvgIcon-colorPrimary",
                                      "colorSecondary": "MuiSvgIcon-colorSecondary",
                                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                      "root": "MuiSvgIcon-root",
                                    }
                                  }
                                  fontSize="inherit"
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                                    focusable="false"
                                    viewBox="0 0 24 24"
                                  >
                                    <path
                                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                                    />
                                  </svg>
                                </ForwardRef(SvgIcon)>
                              </WithStyles(ForwardRef(SvgIcon))>
                            </ForwardRef(StarIcon)>
                          </span>
                        </IconContainer>
                      </span>
                      <span
                        key="5"
                      >
                        <IconContainer
                          className="MuiRating-icon MuiRating-iconFilled"
                          value={5}
                        >
                          <span
                            className="MuiRating-icon MuiRating-iconFilled"
                          >
                            <ForwardRef(StarIcon)
                              fontSize="inherit"
                            >
                              <WithStyles(ForwardRef(SvgIcon))
                                fontSize="inherit"
                              >
                                <ForwardRef(SvgIcon)
                                  classes={
                                    Object {
                                      "colorAction": "MuiSvgIcon-colorAction",
                                      "colorDisabled": "MuiSvgIcon-colorDisabled",
                                      "colorError": "MuiSvgIcon-colorError",
                                      "colorPrimary": "MuiSvgIcon-colorPrimary",
                                      "colorSecondary": "MuiSvgIcon-colorSecondary",
                                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                      "root": "MuiSvgIcon-root",
                                    }
                                  }
                                  fontSize="inherit"
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                                    focusable="false"
                                    viewBox="0 0 24 24"
                                  >
                                    <path
                                      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                                    />
                                  </svg>
                                </ForwardRef(SvgIcon)>
                              </WithStyles(ForwardRef(SvgIcon))>
                            </ForwardRef(StarIcon)>
                          </span>
                        </IconContainer>
                      </span>
                    </span>
                  </ForwardRef(Rating)>
                </WithStyles(ForwardRef(Rating))>
              </div>
            </div>
          </a>
        </LinkAnchor>
      </Link>
    </SimilarProductsItem>
  </Router>
</BrowserRouter>
`;
